AzureStack/2015-04-01/swagger/GalleryAdmin.json

```diff
@@ -0,0 +1,502 @@
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ResourceDefinition"
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+     "GalleryItemUriPayload": {
+      "description": "Gallery item URI, Points to the http or https uri where the galery package is located, typically a blob uri",
+      "type": "object",
+      "properties": {
+        "GalleryItemUri": {
+          "description": "Gets or sets the gallery item URI.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourcePlan": {
+      "type": "object",
+      "description": "The reosurce plan object",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "The name of the plan",
+          "readOnly": true
+        },
+        "Product": {
+          "type": "string",
+          "description": "The product code of the plan",
+          "readOnly": true
+        },
+        "PromotionCode": {
+          "type": "string",
+          "description": "The promotion code of the plan",
+          "readOnly": true
+        },
+        "Publisher": {
+          "type": "string",
+          "description": "The publisher of the plan",
+          "readOnly": true
+        }
+      }
+    },
+    "ResourceSku": {
+      "type": "object",
+      "description": "The resource sku object",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "The name of the sku",
+          "readOnly": true
+        },
+        "Tier": {
+          "type": "string",
+          "description": "The tier of the sku",
+          "readOnly": true
+        },
+        "Size": {
+          "type": "string",
+          "description": "The size of the sku",
+          "readOnly": true
+        },
+        "Family": {
+          "type": "string",
+          "description": "The family of the sku",
+          "readOnly": true
+        },
+        "Capacity": {
+          "format": "int32",
+          "type": "integer",
+          "description": "The capacity of the sku",
+          "readOnly": true
+        }
+      }
+    },
+    "ResourceDefinition": {
+      "type": "object",
+      "description": "The resource definition object of the azure resource manager",
+      "properties": {
+        "Id": {
+          "type": "string",
+          "description": "The ID of the resource"
+        },
+        "Name": {
+          "type": "string",
+          "description": "The name of the resource",
+          "readOnly": true
+        },
+        "Type": {
+          "type": "string",
+          "description": "The type of the resource",
+          "readOnly": true
+        },
+        "Location": {
+          "type": "string",
+          "description": "The location of the resource"
+        },
+        "Plan": {
+          "$ref": "#/definitions/ResourcePlan",
+          "description": "The plan of the resource"
+        },
+        "Sku": {
```

Comment:
Resource should only have id, name, type, location, tags as its properties. Other properties must be defined in the child class that inherits from the Resource.